### PR TITLE
Fix freeProxyCalls and add comment in list.c

### PR DESCRIPTION
**Pull request by ChungHsuanChen**
In test.c ,when list_destroy greater than 2 times and doing assert , it won't have correct freeProxyCalls.

Test_bug

Therefore, every time initialize freeProxyCalls is essential.
Add Comment about parameter in list.c.

### CODE REVIEW

File: src/list.c [hunk 1]
```diff
@@ -26,6 +26,7 @@ list_new(void) {
 
 /*
  * Free the list.
+   @self: Pointer to the list
```
Comment (stephenmathieson) [inline src/list.c:29]: Please keep comment style consistent.

```suggestion
 * @self: Pointer to the list 
```

The rest of the comments should be updated to match the above.

**Review by stephenmathieson — CHANGES_REQUESTED** [review]
Code style is off.

Also, I'm not sure what this is fixing. The changes here only appear to be whitespace and comments.

**Comment by ChungHsuanChen** [comment]
OK, I have changed it.
The code I fixed is about if static variable freeProxyCalls aren't initialized after destroyed the list , we will get wrong answer when we continuously create and destroy a list(I have wrote a function named test_list_destroy_complexver()).
I will push again after refining code style.
Thanks.

File: src/list.c (hunk 1 above)
Comment (ChungHsuanChen) [reply src/list.c]: Finish Refinement, And I also commited it.

**Comment by nouwaarom** [comment]
@ChungHsuanChen Thanks for improving the tests a bit! Can you please rebase this branch so it can be merged?

**Comment by ChungHsuanChen** [comment]
Ok, I have rebased it.

**Comment by nouwaarom** [comment]
Thanks for your contribution!

**Pull request merged**